.github/workflows/debug_asan_instrumentation_clang.yaml

```diff
@@ -0,0 +1,119 @@
+        type: string
+
+concurrency:
+  # Update of a developer branch cancels the previously scheduled workflow
+  # run for this branch. However, the 'master' branch, release branch, and
+  # tag workflow runs are never canceled.
+  #
+  # We use a trick here: define the concurrency group as 'workflow run ID' +
+  # 'workflow run attempt' because it is a unique combination for any run.
+  # So it effectively discards grouping.
+  #
+  # Important: we cannot use `github.sha` as a unique identifier because
+  # pushing a tag may cancel a run that works on a branch push event.
+  group: ${{ (
+    github.ref == 'refs/heads/master' ||
+    startsWith(github.ref, 'refs/heads/release/') ||
+    startsWith(github.ref, 'refs/tags/')) &&
+    format('{0}-{1}', github.run_id, github.run_attempt) ||
+    format('{0}-{1}-debug-asan-clang', github.workflow, github.ref) }}
+  cancel-in-progress: true
+
+jobs:
+  debug_asan_instrumentation_clang:
+    # Run on push to the 'master' and release branches of tarantool/tarantool
+    # or on pull request if the 'full-ci' or 'asan-ci' label is set.
+    if: github.repository_owner == 'tarantool' &&
+        ( github.event_name != 'pull_request' ||
+          contains(github.event.pull_request.labels.*.name, 'full-ci') ||
+          contains(github.event.pull_request.labels.*.name, 'asan-ci') )
+
+    runs-on: ubuntu-20.04-self-hosted
+
+    timeout-minutes: 60
+
+    container:
+      image: docker.io/tarantool/testing:ubuntu-jammy-clang16
+      # Mount /dev to the container to be able to mount a disk image inside it
+      # for successful run of the .github/actions/environment action.
+      volumes:
+        - /dev:/dev
+      # Our testing expects that the init process (PID 1) will
+      # reap orphan processes. At least the following test leans
+      # on it: app-tap/gh-4983-tnt-e-assert-false-hangs.test.lua.
+      # Add extra privileges to the container for successful run
+      # of the .github/actions/environment action.
+      options: '--init --privileged'
+
+    steps:
+      - name: Prepare checkout
+        uses: tarantool/actions/prepare-checkout@master
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+          repository: tarantool/tarantool
+          ref: ${{ inputs.submodule && 'master' || github.ref }}
+      - uses: ./.github/actions/environment
+      - name: Install deps
+        uses: ./.github/actions/install-deps-debian
+      - name: Optional submodule bump
+        if: ${{ inputs.submodule }}
+        uses: ./.github/actions/bump-submodule
+        with:
+          submodule: ${{ inputs.submodule }}
+          revision: ${{ inputs.revision }}
+      - name: Switch luajit submodule branch
+        run: |
+            cd third_party/luajit
+            git fetch origin mandesero/dlmalloc-instr
+            git checkout mandesero/dlmalloc-instr
+            cd ../..
+      - name: test
+        env:
+          CC: clang-16
```

Comment:
Why Clang 16? Clang 18 [^1] is available in LLVM APT repository [^2].

[^1]: https://releases.llvm.org/18.1.0/tools/clang/docs/ReleaseNotes.html
[^2]: https://apt.llvm.org/

Reply:
`clang-16` is used in existing tests (`debug/release_asan_clang.yaml`), so I did not replace it with `clang-18`.

Reply:
> Addressed in PR #10605.

I have rebased this PR so it now uses clang-19.